scripts/S3_FSCK/s3_fsck_p0.py

```diff
@@ -146,25 +157,61 @@ def blob(row):
+                    # digkey: the unique part of a main chunk before service id,
+                    # arc schema, and class are appended
                     rtlst.append(
                         {"key": key, "subkey": k, "digkey": gen_md5_from_id(k)[:26]}
                     )
+                # If the key is split and request is OK:
+                # return a list of dicts with the key (primary sproxyd input key),
+                # subkey (sproxyd input key of a split stripe) and
+                # digkey, (md5 of the subkey)
+                # digkey: the unqiue part of a main chunk before service id,
+                # arc schema, and class are appended
                 return rtlst
+            # If the key is split and request is not OK:
+            # return a dict with the key (primary sproxyd input key)
+            # with both subkey and digkey columns set to NOK
             return [{"key": key, "subkey": "NOK", "digkey": "NOK"}]
         except requests.exceptions.ConnectionError as e:
+            # If there is a Connection Error in the HTTP request:
+            # return a dict with the key(primary sproxyd input key),
+            # with both subkey and digkey set to NOK
             return [{"key": key, "subkey": "NOK_HTTP", "digkey": "NOK_HTTP"}]
     if not split['is_split']:
+        # If the key is not split:
+        # return a dict with the key (primary sproxyd input key),
+        # subkey set to SINGLE and
+        # digkey, (md5 of the subkey)
+        # digkey: the unique part of a main chunk before service id,
+        # arc schema, and class are appended
         return [{"key": key, "subkey": "SINGLE", "digkey": gen_md5_from_id(key)[:26]}]
 
 new_path = os.path.join(PATH, RING, "s3-bucketd")
 files = "%s://%s" % (PROTOCOL, new_path)
 
+# reading without a header,
+# columns _c0, _c1, _c2 are the default column names of
+# columns   1,   2,   3 for the csv
+# input structure: (bucket name, s3 object key, sproxyd input key)
+#   e.g. test,48K_object.01,9BC9C6080ED24A42C2F1A9C78F6BCD5967F70220
+# Required Fields:
+#   - _c2 (sproxyd input key)
 df = spark.read.format("csv").option("header", "false").option("inferSchema", "true").option("delimiter", ",").load(files)
 
+# repartition the dataframe to have the same number of partitions as the number of executors * cores
 df = df.repartition(PARTITIONS)
+# Return a new Resilient Distributed Dataset (RDD) by applying a function to each element of this RDD.
 rdd = df.rdd.map(lambda x : blob(x))
+# Return a new RDD by first applying a function to all elements of this RDD, and then flattening the results. Then transform it into a dataframe.
 dfnew = rdd.flatMap(lambda x: x).toDF()
 
 single = "%s://%s/%s/s3fsck/s3-dig-keys.csv" % (PROTOCOL, PATH, RING)
+# write the dataframe to a csv file with a header
+#   output structure: (digkey, sproxyd input key, subkey if available)
 dfnew.write.format("csv").mode("overwrite").options(header="true").save(single)
```

Comment:
If we want correct headers, this write operation w/ `header="true"` is where we get our first instance of generic **_c0**, **_c1** column names. We could perform:
```suggestion
dfnew = dnew.withColumnRenamed("_c0", "digkey).withColumnRenamed("_c1", "input_key").withColumnRenamed("_c2", "subkey")
dfnew.write.format("csv").mode("overwrite").options(header="true").save(single)
```

Requires updating the p2 script to read the new column names instead of the generic ones.